tests/functional/unit_testing/test_ut_versions.py

```diff
@@ -0,0 +1,162 @@
+import pytest
+from dbt.tests.util import run_dbt, get_unique_ids_in_results, write_file
+from dbt.exceptions import YamlParseDictError, ParsingError
+
+from tests.functional.unit_testing.fixtures import (
+    my_model_versioned_yml,
+    my_model_versioned_no_2_yml,
+    test_my_model_all_versions_yml,
+    test_my_model_exclude_versions_yml,
+    test_my_model_include_versions_yml,
+    test_my_model_include_exclude_versions_yml,
+    test_my_model_include_unversioned_yml,
+    test_my_model_version_ref_yml,
+    my_model_v1_sql,
+    my_model_v2_sql,
+    my_model_v3_sql,
+    my_model_a_sql,
+    my_model_b_sql,
+    my_model_sql,
+    my_model_version_ref_sql,
+)
+
+
+# test with no version specified, then add an exclude version, then switch
+# to include version and make sure the right unit tests are generated for each
+class TestVersions:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model_a.sql": my_model_a_sql,
+            "my_model_b.sql": my_model_b_sql,
+            "my_model_v1.sql": my_model_v1_sql,
+            "my_model_v2.sql": my_model_v2_sql,
+            "my_model_v3.sql": my_model_v3_sql,
+            "schema.yml": my_model_versioned_yml,
+            "unit_tests.yml": test_my_model_all_versions_yml,
+        }
+
+    def test_versions(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 5
+
+        # "my_model" has three versions: 1, 2, 3
+        # There is a single unit_test which doesn't specify a version,
+        # so it should run for all versions.
+        results = run_dbt(["test"])
+        assert len(results) == 3
+        unique_ids = get_unique_ids_in_results(results)
+        expected_ids = [
+            "unit_test.test.my_model.test_my_model_v1",
+            "unit_test.test.my_model.test_my_model_v2",
+            "unit_test.test.my_model.test_my_model_v3",
+        ]
+        assert sorted(expected_ids) == sorted(unique_ids)
+
+        # Select tests for a single versioned model
+        results = run_dbt(["test", "--select", "my_model.v2"])
```

Comment:
Should we also be able to select tests for all versions and expect all unit tests for each version of the model?

`run_dbt(["test", "--select", "my_model"])`

Reply:
Yes, that should work. I can make another test for it.

Reply:
Done.